frame/support/procedural/src/pallet/expand/genesis_config.rs

```diff
@@ -94,15 +94,24 @@ pub fn expand_genesis_config(def: &mut Def) -> proc_macro2::TokenStream {
 					"]
 				));
 			}
-			attrs.push(syn::parse_quote!( #[cfg(feature = "std")] ));
 			attrs.push(syn::parse_quote!(
-				#[derive(#frame_support::Serialize, #frame_support::Deserialize)]
+				#[cfg_attr(feature = "std", derive(#frame_support::Serialize, #frame_support::Deserialize))]
 			));
-			attrs.push(syn::parse_quote!( #[serde(rename_all = "camelCase")] ));
-			attrs.push(syn::parse_quote!( #[serde(deny_unknown_fields)] ));
-			attrs.push(syn::parse_quote!( #[serde(bound(serialize = ""))] ));
-			attrs.push(syn::parse_quote!( #[serde(bound(deserialize = ""))] ));
-			attrs.push(syn::parse_quote!( #[serde(crate = #serde_crate)] ));
+			attrs.push(
+				syn::parse_quote!( #[cfg_attr(feature = "std", serde(rename_all = "camelCase"))] ),
+			);
+			attrs.push(
+				syn::parse_quote!( #[cfg_attr(feature = "std", serde(deny_unknown_fields))] ),
+			);
+			attrs.push(
+				syn::parse_quote!( #[cfg_attr(feature = "std", serde(bound(serialize = "")))] ),
+			);
+			attrs.push(
+				syn::parse_quote!( #[cfg_attr(feature = "std", serde(bound(deserialize = "")))] ),
+			);
+			attrs.push(
+				syn::parse_quote!( #[cfg_attr(feature = "std", serde(crate = #serde_crate))] ),
```

Comment:
Maybe you can push them all in one `quote` call?

Reply:
Indeed, this is possible through binding a new `TokenStream` called `serde_attr_arg` or something similar, and use the `quote!` macro for it.

Reply:
this is `parse_quote!` and will not work.